Code/GraphMol/Chirality.cpp

```diff
@@ -1089,6 +1088,88 @@ void buildCIPInvariants(const ROMol &mol, DOUBLE_VECT &res) {
   }
 }
 
+//! Lightweight sortable wrapper that references a CIP entry and keeps track of
+//! the current rank.
+struct SortableCIPReference {
+  SortableCIPReference(CIP_ENTRY* cipRef, const int atomIdx): cip(cipRef), atomIdx(atomIdx) {}
+  SortableCIPReference(SortableCIPReference && other) noexcept {
+    cip = other.cip;
+    atomIdx = other.atomIdx;
+    other.cip = nullptr;
+    currRank = other.currRank;
+  }
+  SortableCIPReference & operator=(SortableCIPReference && other) noexcept {
+    cip = other.cip;
+    atomIdx = other.atomIdx;
+    other.cip = nullptr;
+    currRank = other.currRank;
+    return *this;
+  }
+
+  bool operator==( const SortableCIPReference & rhs ) const {
```

Comment:
If there were any chance that this code was being used in non-controlled situations, this (and the next function) would want either a PRECONDITION that `cip` here and in `rhs` is not null or some logic to handle that case.
It's probably not worth adding here now though

Reply:
Done anyway